### ISSUE TITLE
 df.copy() does not perform shallow copy

### ISSUE THREAD

**Issue by AK-ayush** [rapidsai/cudf#327]
## Reporting a bug

- [x] I am using the latest version of cuDF from conda, built from `master`, or
built from the latest tagged release. 
- [ ] I have included the version or commit hash I am using for reference.
- [ ] I have included the following environment details:
      Linux Distro, Linux Kernel, GPU Model
- [ ] I have included the following version information for:
      Arrow, cmake, CUDA, gcc/g++, Numpy, Pandas, Python
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

[`copy`](https://github.com/rapidsai/cudf/blob/master/cudf/dataframe.py#L357) method of cudf DataFrame is performing deep copy while docstring written in method says **"shallow copy of dataframe"**.  
Here is minimal reproducer:
```
from cudf.dataframe import DataFrame
df = DataFrame()
df['a']=[1,2,3] 
df2 = df.copy()
df2['a'] = [4,5,6]
```
```
print(df)                                                                                                                                                             
     a
0    1
1    2
2    3
```
```
print(df2)                                                                                                                                                            
     a
0    4
1    5
2    6
```
Please, correct me if my understanding of shallow and deep copy is wrong.

**Comment by thomcom** [comment]
WIP coming up

**Comment by thomcom** [comment]
This is fixed in PR #600. More accurately, PR #600 causes us to properly emulate Pandas copying - `df.copy()` defaults to deep. The only way to get a shallow copy in pandas is with `df.copy(deep=False)` - all other methods default to deep.

**Comment by kkraus14** [comment]
Fixed by #600.
